File: fapi/src/main/java/com/artsykov/fapi/models/CustomerOrCompanyOrAdminOrErrorsModel.java
package com.artsykov.fapi.models;

import lombok.Data;

import java.util.Map;

@Data
public class CustomerOrCompanyOrAdminOrErrorsModel {
    private CustomerModel customerModel;
    private CompanyModel companyModel;
    private AdminModel adminModel;
    private Map<String, String> errors;

    public CustomerOrCompanyOrAdminOrErrorsModel() {}

    public CustomerOrCompanyOrAdminOrErrorsModel(Map<String, String> errors) {
        this.errors = errors;
    }
}

File: frontend/src/app/modules/models/customerOrCompanyOrAdminOrErrorsModel.ts
import {customerModel} from "./customerModel";
import {companyModel} from "./companyModel";
import {adminModel} from "./adminModel";

export class customerOrCompanyOrAdminOrErrorsModel {
  customerModel: customerModel = new customerModel();
  companyModel: companyModel = new companyModel();
  adminModel: adminModel = new adminModel();
  errors: Map<string, string> = new Map<string, string>();
}
